### PR TITLE
Add secret to password hashing

### PR DESCRIPTION
**Pull request by zkhin**
Fixes: https://github.com/SeedCompany/cord-api-v3/issues/1217

### CODE REVIEW

File: src/core/config/config.service.ts [hunk 1]
```diff
@@ -91,6 +91,13 @@ export class ConfigService {
     };
   }
 
+  @Lazy() get passwordSecret() {
+    return Buffer.from(
+      this.env.string('PASSWORD_SECRET').optional('passwordSecret'),
```
Comment (CarsonF) [inline src/core/config/config.service.ts:96]: Let's just omit it by default as to not mess up existing databases
```suggestion
      this.env.string('PASSWORD_SECRET').optional(),
```

File: src/core/config/config.service.ts [hunk 2]
```diff
@@ -91,6 +91,13 @@ export class ConfigService {
     };
   }
 
+  @Lazy() get passwordSecret() {
+    return Buffer.from(
```
Comment (CarsonF) [inline src/core/config/config.service.ts:95]: Would you mind doing this conversion has a private getter/method in the AuthenticationService? I don't think the fact that argon2 needs it as a buffer should concern the config service.

**Review by CarsonF — CHANGES_REQUESTED** [review]

**Review by CarsonF — DISMISSED** [review]
Thanks @zkhin!

File: src/components/authentication/authentication.service.ts [hunk 3]
```diff
@@ -419,9 +420,11 @@ export class AuthenticationService {
...
 
-  private get argon2Options() {
-    return this.config.passwordSecret
-      ? { secret: Buffer.from(this.config.passwordSecret, 'utf-8') }
-      : {};
+  private get argon2Options(): Except<argon2.Options, 'raw'> {
+    return {
+      secret: this.config.passwordSecret
+        ? Buffer.from(this.config.passwordSecret, 'utf-8')
+        : undefined,
```
Comment (zkhin) [inline src/components/authentication/authentication.service.ts:427]: I don't think `undefined` works for argon2.Options

Comment (CarsonF) [reply src/components/authentication/authentication.service.ts:427]: Really? Ok good catch.

**Review by CarsonF — APPROVED** [review]

**Pull request merged**